tools/killswitch/src/output.rs

```diff
@@ -0,0 +1,209 @@
+                }
+            });
+            (channel.clone(), (false, (channel.replica, val)))
+        })
+        .collect::<Vec<(_, (_, _))>>();
+    replicas.extend(good.into_iter().map(|(channel, tx)| {
+        let val = json!({
+            "result": {
+                "status": "success",
+                "tx_hash": format!("{:?}", tx.txid),
+                "message": serde_json::Value::Null,
+            }
+        });
+        (channel.clone(), (true, (channel.replica, val)))
+    }));
+    let mut homes: HashMap<String, Vec<(bool, (String, Value))>> = HashMap::new();
+    for (channel, (success, replica)) in replicas {
+        if let Some(replicas) = homes.get_mut(&channel.home) {
+            replicas.push((success, replica));
+        } else {
+            homes.insert(channel.home, vec![(success, replica)]);
+        }
+    }
+    Message::FullMessage(json!({
+        "homes": homes.into_iter().map(|(home, replicas)| {
+            // report error for *any* errors encountered
+            let success = replicas.iter().all(|(s, _)| *s);
+            (home, json!({
+                "status": if success { "success" } else { "error" },
+                "message": {
+                    "replicas": replicas
+                        .into_iter()
+                        .map(|(_, replica)| replica)
+                        .collect::<HashMap<String, Value>>(),
+                }
+            }))
+        }).collect::<HashMap<String, Value>>()
+    }))
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use ethers::core::types::H256;
+    use nomad_core::TxOutcome;
+    use std::str::FromStr;
+
+    #[test]
+    fn it_produces_correct_bad_output() {
+        let channel1 = Channel {
+            home: "ethereum".into(),
+            replica: "avalanche".into(),
+        };
+        let channel2 = Channel {
+            home: "avalanche".into(),
+            replica: "ethereum".into(),
+        };
+        let error1 = Error::MissingRPC(channel1.home.clone());
+        let error2 = Error::MissingAttestationSignerConf(channel1.home.clone());
+        let error3 = Error::MissingTxSubmitterConf(channel2.replica.clone());
+        let bad = vec![(channel1, vec![error1, error2]), (channel2, vec![error3])];
+        let json = match build_output_message(bad, vec![]) {
+            Message::FullMessage(json) => json,
+            _ => panic!("Match error. Should never happen"),
+        };
+        let json = serde_json::to_string(&json).unwrap();
+
+        let value: Value = serde_json::from_str(&json).unwrap();
+        let ethereum = &value["homes"]["ethereum"];
+        let avalanche = &value["homes"]["avalanche"];
+        assert_eq!(ethereum["status"], "error");
+        assert_eq!(avalanche["status"], "error");
+        assert_eq!(
+            &avalanche["message"]["replicas"]["ethereum"]["result"]["message"][0],
+            "MissingTxSubmitterConf: No transaction submitter config found for: ethereum"
```

Comment:
Ditto RE concrete types instead of stringly typed JSON or `serde::Value`. We could make assertions easier if we deserialized the JSON output back into concrete type